File: backend/count_users_detailed.py
import sys
import os
from flask_mongoengine import MongoEngine
from flask import Flask
from dotenv import load_dotenv

# Load env vars
load_dotenv()

app = Flask(__name__)
app.config['MONGODB_SETTINGS'] = {
    'host': os.getenv('MONGO_URI')
}
db = MongoEngine(app)

class PendingUser(db.Document):
    meta = {'collection': 'pending_users'}
    email = db.StringField()

class User(db.Document):
    meta = {'collection': 'users'}
    email = db.StringField()

def count_users():
    with app.app_context():
        user_count = User.objects.count()
        pending_count = PendingUser.objects.count()
        print(f"Total Users: {user_count}")
        print(f"Total Pending Users: {pending_count}")

if __name__ == "__main__":
    count_users()

File: backend/src/scripts/countUsersDetailed.js
/**
 * scripts/countUsersDetailed.js
 * Counts total Users and PendingUsers.
 */

import dotenv from 'dotenv';
import connectDB from '../config/db.js';
import { User, PendingUser } from '../models/index.js';
import mongoose from 'mongoose';

dotenv.config();

async function main() {
  await connectDB();
  const userCount = await User.countDocuments();
  const pendingCount = await PendingUser.countDocuments();
  console.log(`Total Users: ${userCount}`);
  console.log(`Total Pending Users: ${pendingCount}`);
  await mongoose.disconnect();
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
